src/manifest.js

```diff
@@ -261,22 +261,37 @@ export const addPropertiesToMaster = (master, uri) => {
   }
 
   forEachMediaGroup(master, (properties, mediaType, groupKey, labelKey) => {
-    if (!properties.playlists ||
-        !properties.playlists.length ||
-        properties.playlists[0].uri) {
-      return;
+    const groupId = `placeholder-uri-${mediaType}-${groupKey}-${labelKey}`;
+
+    if (!properties.playlists || !properties.playlists.length) {
+      properties.playlists = [Object.assign({}, properties)];
     }
 
-    // Set up phony URIs for the media group playlists since playlists are referenced by
-    // their URIs throughout VHS, but some formats (e.g., DASH) don't have external URIs
-    const phonyUri = `placeholder-uri-${mediaType}-${groupKey}-${labelKey}`;
-    const id = createPlaylistID(0, phonyUri);
+    properties.playlists.forEach(function(p, i) {
+      const id = createPlaylistID(i, groupId);
+
+      // DEPRECATED, this has been added to prevent a breaking change.
+      // previously we only ever had a single media group playlist, so
+      // we mark the first playlist uri without prepending the index as we used to
+      // ideally we would do all of the playlists the same way.
+      if (i === 0) {
+        p.uri = p.uri || groupId;
+      } else {
+        p.uri = p.uri || id;
+      }
+
+      p.id = p.id || id;
+
+      p.resolvedUri = p.resolvedUri || resolveUrl(master.uri, p.uri);
```

Comment:
If the playlist doesn't have a URI, would this lead to a weird result (given that it would use the ID)?

Reply:
It will give us weird results, but there are a few places in hls that expect a resolved uri so our media playlist needs to have one.

Reply:
Should we just use the ID as-is in those cases? Might be easier to understand as a property if it's not an actual URI (that doesn't exist).